src/data/index.ts

```diff
@@ -48,23 +48,100 @@ export type {
   QueryShared,
 } from './query';
 
+/**
+ * @typeParam T - The type of metadata associated with each record.
+ */
 export class Index<T extends RecordMetadata = RecordMetadata> {
   private config: PineconeConfiguration;
   private target: {
     index: string;
     namespace: string;
   };
 
+  /**
+   * Delete all records from the index.
+   *
+   * @example
```

Comment:
I think the example should demonstrate that only the targeted namespace is affected.

```typescript
const pinecone = new Pinecone()
await client.index('my-index').describeIndexStats()
// {
//   namespaces: { 
//     '': { recordCount: 9 }, 
//     foo: { recordCount: 1 } 
//   },
//   dimension: 8,
//   indexFullness: 0,
//   totalRecordCount: 10
// }

await client.index('my-index').deleteAll()

// Records in default namespace '' are now gone, but
// records in namespace 'foo' are not modified.
await client.index('my-index').describeIndexStats()
// {
//   namespaces: { foo: { recordCount: 1 } },
//   dimension: 8,
//   indexFullness: 0,
//   totalRecordCount: 1
// }
```

Reply:
Agreed, I think it's definitely helpful to have more examples involving namespace to make things more explicit.